ivi-layermanagement-api/test/ilm_check_env.cpp

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+ *
+ * Copyright (C) 2023 Advanced Driver Information Technology Joint Venture GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <gtest/gtest.h>
+#include <wayland-client.h>
+#include "ivi-input-client-protocol.h"
+#include "ivi-wm-client-protocol.h"
+
+class TestEnvChecking
+{
+  public:
+    struct wl_display  *mpDisplay;
+    struct wl_registry *mpRegistry;
+
+    struct ivi_wm *mpController;
+    struct ivi_input *mpInputController;
+    bool mCheck;
+
+    static constexpr uint32_t IVI_CONTROLLER_VERSION{1U};
+    static constexpr uint32_t IVI_INPUT_VERSION{2U};
+
+    static TestEnvChecking *GetInstance();
+    ~TestEnvChecking();
+
+  private:
+    static void RegistryHandleGlobal(void *apData, struct wl_registry *apRegistry, uint32_t aId, const char *acpInterface, uint32_t aVersion);
+    static constexpr struct wl_registry_listener mRegistryListener = {
+      RegistryHandleGlobal,
+      nullptr
+    };
+
+    static TestEnvChecking *mpInstance;
+    TestEnvChecking();
+};
+
+TestEnvChecking* TestEnvChecking::mpInstance{nullptr};
+
+TestEnvChecking::TestEnvChecking():mpDisplay(nullptr), mpRegistry(nullptr), mpController(nullptr), mpInputController(nullptr), mCheck(true)
+{
+  int ret;
+
+  mpDisplay = wl_display_connect(nullptr);
+  if (mpDisplay == nullptr) {
+    printf("ERROR: Failed to wl_display_connect()\n");
+    mCheck = false;
+    return;
+  }
+
+  mpRegistry = wl_display_get_registry(mpDisplay);
+  if (mpRegistry == nullptr) {
+    printf("ERROR: Failed to wl_display_get_registry()\n");
+    mCheck = false;
+    return;
+  }
+
+  ret = wl_registry_add_listener(mpRegistry, &mRegistryListener, this);
+  if (ret < 0) {
+    printf("ERROR: Failed to wl_registry_add_listener()\n");
+    mCheck = false;
+    return;
+  }
+
+  ret = wl_display_dispatch(mpDisplay);
+  if (ret < 0) {
+    printf("ERROR: Failed to wl_display_dispatch()\n");
+    mCheck = false;
+    return;
+  }
+
+  if ((mpController == nullptr) || (mpInputController == nullptr))
+  {
+    printf("ERROR: can't get the ivi_wm (%lX) or ivi_input (%lX) protocols from compositor.\n", reinterpret_cast<uintptr_t>(mpController), reinterpret_cast<uintptr_t>(mpInputController));
```

Comment:
printf("ERROR: can't get the ivi_wm (%lX) or ivi_input (%lX) protocols from compositor.\n", 
        reinterpret_cast<uintptr_t>(mpController),
        reinterpret_cast<uintptr_t>(mpInputController));

Reply:
Thank @efriedrich, I updated it